### DISCUSSION TITLE
How can I add a segment which inherits the background color of the previous one?

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [JanDeDobbeleer/oh-my-posh#988] [Troubleshoot] [answered]
I made my shell line end with a rounding character (`\uE0B6` for start, `\uE0B4` for end):

![my current shell line, rounded off at the start and end](https://user-images.githubusercontent.com/65342367/134552889-c5810825-f6d2-4748-aa34-cab8c1de3376.png)

But now I want it to basically round off whatever segment is currently at the end, basically meaning I want it to inherit the last segment's background color (in the block).

Is there a way to do that?

**Comment by lnu** [comment]
Can you put a copy of your theme?
if your last segment is a diamond segment you can put a custom [trailing_diamond](https://ohmyposh.dev/docs/configure#trailing-diamond):
```
{
          "background": "#2e9599",
          "foreground": "#ffffff",
          "leading_diamond": "",
          "properties": {
            "always_enabled": true,
            "color_background": true,
            "display_exit_code": false,
            "error_color": "#f1184c",
            "prefix": " "
          },
          "style": "diamond",
          "trailing_diamond": "\uE0B4",
          "type": "exit"
}
```
gives:
![image](https://user-images.githubusercontent.com/1829553/134685388-ab3b1174-aba2-4b1d-898f-60f8469b44a1.png)

But from what you explain you'd like a fixed segment at the end which takes the color of any previous segment?

**Reply by JanDeDobbeleer** [reply]
Yes and we don't have that easily out of the box tbh. But, I like the `inherit` color option idea.

**Reply by TheOnlyTails** [reply]
Having an `inherit` color which goes to the previous segment (you could even have a block option that specifies whether `inherit` is allowed to inherit from other blocks) would be nice.
Would you like me to open an issue?

**Reply by JanDeDobbeleer** [reply]
Please do, I believe this can be added rather swiftly.

**Reply by TheOnlyTails** [reply]
Done! #994

**Reply by TheOnlyTails** [reply]
I don't know how to use Go, but otherwise I'd love to help actually implement it.

**Comment by TheOnlyTails** [comment, marked as answer]
Looks amazing! I'm very impressed with how fast this was done, @JanDeDobbeleer!

![image](https://user-images.githubusercontent.com/65342367/134720072-c83474f4-bfe2-4228-b146-c1f14ec7d76c.png)
![image](https://user-images.githubusercontent.com/65342367/134720100-0af6f8e0-e4e9-4260-905e-1a144edb9354.png)

```json5
{
	"type": "text",
	"style": "diamond",
	"background": "inherit",
	"foreground": "inherit",
	"trailing_diamond": "\uE0B4", // this makes the end round
	"properties": {
		"text": ""
	}
}
```

**Reply by JanDeDobbeleer** [reply]
My pleasure.
